Formula/h/hiredict.rb

```diff
@@ -0,0 +1,26 @@
+class Hiredict < Formula
+  desc "C client library for Redict"
+  homepage "https://codeberg.org/redict/hiredict"
+  url "https://codeberg.org/redict/hiredict/archive/1.3.1.tar.gz"
+  sha256 "377eb938930825a07b5b48c6dd171c59f439cd8c87c33f9c8dff6da3fc11ff5a"
+  license all_of: ["BSD-3-Clause", "LGPL-3.0-or-later"]
+  head "https://codeberg.org/redict/hiredict.git", branch: "main"
+
+  depends_on "openssl@3"
+
+  def install
+    system "make", "install", "PREFIX=#{prefix}", "USE_SSL=1"
+    pkgshare.install "examples"
+
+    # remove versioned dynamic library links
```

Comment:
Why?

Reply:
it caused some audit failure

```
hiredict
  * Non-libraries were installed to "/opt/homebrew/opt/hiredict/lib".
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /opt/homebrew/opt/hiredict/lib/libhiredict.dylib.1
      /opt/homebrew/opt/hiredict/lib/libhiredict_ssl.dylib.1.3.1
      /opt/homebrew/opt/hiredict/lib/libhiredict_ssl.dylib.1
```

Reply:
The file format is wrong. They should be `libhiredict_ssl.1.3.1.dylib`

Reply:
Seems to be an upstream problem. Version suffixes for dylibs on macOS come before the `.dylib` extension name (`libhiredict.1.3.1.dylib`), which is different from other systems (`libhiredict.so.1.3.1` on Linux).

Working on a fix.

Reply:
Looks like an upstream bug. https://codeberg.org/redict/hiredict/src/commit/57291e2244ebb6c32784cd81244ba88d0e53b8b9/Makefile#L157 sets `DYLIB_MINOR_NAME` but they forgot to also set `DYLIB_MAJOR_NAME` similarly (like https://codeberg.org/redict/hiredict/src/commit/57291e2244ebb6c32784cd81244ba88d0e53b8b9/Makefile#L65).

Neither `SSL_DYLIB_MINOR_NAME` or `SSL_DYLIB_MAJOR_NAME` is set correctly either.

Reply:
Proposed fix: https://codeberg.org/redict/hiredict/commit/73e0f09f8ad89279ff116d5fb49ff2211c56f5f1.patch

Upstream PR: https://codeberg.org/redict/hiredict/pulls/16

Reply:
Thanks @ZhongRuoyu! 💪